Implement Redis v6 auth (fixes #182)

jessie-murray opened the pull request:
There are 3 commits in this PR:
1. 4640ac0 adds support for Redis 6 auth with a username and password
2. f159c7f fixes an issue I noticed in `slog.c` while writing this
3. d3d40bd adds logging for the `AUTH` command; webdis is currently ignoring the response from Redis and just relying on later commands failing to notify users that the current config doesn't work correctly

CC @nicolasff

nicolasff reviewed (changes requested):
This looks great! Thanks so much for the PR. I think there is just one issue with the way an error is logged when auth is disabled, but otherwise all my other comments were minor suggestions.

jessie-murray commented on src/conf.c:
```diff
@@ -294,3 +308,47 @@ conf_free(struct conf *conf) {
+		return NULL;
+	}
+	json_t *jusername = json_array_get(jarray, 0);
+	json_t *jpassword = json_array_get(jarray, 1);
```
I see what you mean but this is not actually the case today, for example in this file itself there are variables declared inside of `if` blocks, like

```c
const char *plain = conf_string_or_envvar(json_string_value(jbasic));
```

jessie-murray commented on src/conf.c:
```diff
@@ -294,3 +308,47 @@ conf_free(struct conf *conf) {
+
+	char *username = conf_string_or_envvar(json_string_value(jusername));
+	char *password = conf_string_or_envvar(json_string_value(jpassword));
+	struct auth *ret = calloc(1, sizeof(struct auth));
```
See response above.

jessie-murray commented on src/conf.c:
```diff
@@ -16,6 +16,14 @@
+static struct auth *
+conf_auth_legacy(char *password);
+static struct auth *
+conf_auth_username_password(json_t *jarray);
```
I started adding some but this adds a lot of noise to the code, I'm not sure how to say it. I find it a lot less readable and it's not super important in a file like this one. Maybe it makes more sense in other places where the code is not executed just one time?

jessie-murray commented:
@nicolasff I added a new commit with the bugfix for the error message logged when auth is disabled, also I updated `conf_free` to fully free the `redis_auth` struct but then I noticed that it is never called since Webdis just exits with `exit(0)` so I was not really sure what to do.
